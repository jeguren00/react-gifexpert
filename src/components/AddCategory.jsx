import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({onAddCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 1) {
            return;
        }
        onAddCategory(inputValue.trim());
        setInputValue('');
    }
    return (
        <form aria-label="form" onSubmit={onSubmit}>
            <input 
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={(event) => onInputChange(event)}
             />
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
  }