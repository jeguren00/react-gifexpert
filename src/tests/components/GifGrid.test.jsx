import { render, screen } from '@testing-library/react';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock("../../hooks/useFetchGifs");


describe('Pruebas en <GifGrid />', () => {
    
    const category = 'One Punch';

    test('debe de mostrar el loading inicialmente', () => {

        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true
        });


        render( <GifGrid category={ category } /> );
        //expect( screen.getByText( 'Cargando...' ) ); //No encuentra el Cargando, creo por los tiempos de ejecucion ya que carga mas rapido la web
        expect( screen.getByText( category ) );

    });

    test('debe de mostrar items cuando se cargan las imÃ¡genes useFetchGifs', () => {
        
        const gifs = [
            {
                id: 'ABC',
                title: 'Saitama',
                url: 'https://localhost/saitama.jpg'
            },
            {
                id: '123',
                title: 'Goku',
                url: 'https://localhost/goku.jpg'
            },
        ]

        useFetchGifs.mockReturnValue({
            images: gifs,
            isLoading: false
        });

        render( <GifGrid category={ category } /> );
        expect( screen.getAllByRole('img').length ).toBe(2);
        


    });


});