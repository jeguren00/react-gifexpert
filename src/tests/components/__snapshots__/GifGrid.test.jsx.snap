// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas gif grid to match snapshot 1`] = `
<div>
  <h3>
    fallout
  </h3>
  <div
    class="card-grid"
  />
</div>
`;
